indico/modules/rb_new/client/js/common/bookings/reducers.js

```diff
@@ -27,13 +27,22 @@ export default combineReducers({
             bookingsActions.FETCH_BOOKING_DETAILS_REQUEST,
             bookingsActions.FETCH_BOOKING_DETAILS_SUCCESS,
             bookingsActions.FETCH_BOOKING_DETAILS_ERROR
-        )
+        ),
+        changePreBookingState: requestReducer(
+            bookingsActions.BOOKING_STATE_CHANGE_REQUEST,
+            bookingsActions.BOOKING_STATE_CHANGE_SUCCESS,
+            bookingsActions.BOOKING_STATE_CHANGE_ERROR
+        ),
     }),
     details: (state = {}, action) => {
         switch (action.type) {
             case bookingsActions.BOOKING_DETAILS_RECEIVED: {
-                const {id, ...data} = camelizeKeys(action.data);
-                return {...state, [id]: data};
+                const data = camelizeKeys(action.data);
+                return {...state, [data.id]: data};
+            }
+            case bookingsActions.BOOKING_STATE_UPDATED: {
```

Comment:
since you return all booking data you can just send dispatch `BOOKING_DETAILS_RECEIVED` instead of adding a new action

Reply:
I don't return everything. If I want to do it in `BOOKING_DETAILS_RECEIVED` I will have to make sure it doesn't override entirely the object that we received from the `BookingDetails` endpoint. I could also just return state-related data and leave the additional action.